### DISCUSSION TITLE
A simple "useReaction" useState-like hook?

### DISCUSSION THREAD

**Discussion by mckravchyk** [mobxjs/mobx#3824] [General] [answered]
Hey. I was wondering, is it possible to just do something like this? To get a hook for an observable value exactly like useState?

```js
const todoEntry = useReaction(() => store.todos.todos.get(id));
```

where store is a value in the context, i.e.:

```js
export function createStore() {
  const todosSlice = {
    todos: observable.map(),
    order: [],
  };

  makeObservable(todosSlice, {
    order: observable,
  });

  return {
    todos: todosSlice,
  };
};

const store = createStore();
```

I look at the mobx-react-lite README.md and it seems like the presented workflow is somewhat different.

**Comment by urugator** [comment, marked as answer]
https://github.com/mobxjs/mobx/discussions/2566#discussioncomment-565121

**Reply by mckravchyk** [reply]
Thanks!
